xarray/core/variable.py

```diff
@@ -647,6 +647,9 @@ def transpose(self, *dims):
         """
         if len(dims) == 0:
             dims = self.dims[::-1]
+        # no need to transpose if only one dimension
+        if len(dims) == 1:
+            return self
```

Comment:
It's OK to keep the data the same, but we shouldn't return the same `Variable` object. I would use `self.copy(deep=False)` instead. Also, let's check *after* we verify that `dims` is valid (e.g., move the conditional one line below). Probably a little safer to use `if self.ndim < 2` as the check instead, too.